File: src/main/java/sen/saloum/JobConnect/model/Job.java
package sen.saloum.JobConnect.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.internal.build.AllowNonPortable;

import java.time.OffsetDateTime;

@Entity
@Data
@NoArgsConstructor
@AllowNonPortable
public class Job {
    @Id
    @GeneratedValue
    private Long id;
    private String title;
    private String description;
    private String employmentType;
    private String location;
    private String jobType;
    private int yearExperience;
    private OffsetDateTime datePosted;
    private double pay;
    private String requirements;
    private String fullDescription;
    private String recruiterCompanyName;
    private String categoryName;
    private String companyLogoUrl;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne
    @JoinColumn(name = "recruteur_id")
    private Recruiter recruiter;

}

File: src/app/interfaces/job.ts
export interface Job {
  id: number;
  title: string;
  description: string;
  employmentType: string;
  location: string;
  jobType: string;
  yearExperience: number;
  datePosted: string; 
  pay: number;
  requirements: string;
  categoryId: number;
  categoryName: string;
  recruiterId: number;
  recruiterCompanyName: string;
  timeAgo?: string;
  bookmarked?: boolean;
  companyLogoUrl?: string;
  logoBgColor?: string;
  logoIcon?: string;
  logoIconColor?: string;
  categoryIcon?: string;
  typeIcon?: string;
  showDetails?: boolean;
}